### DISCUSSION TITLE
__init__() got an unexpected keyword argument '_metadata'

### DISCUSSION THREAD

**Discussion by pranayvadd** [NVIDIA-NeMo/NeMo#5373] [Q&A]
Hi Everyone,

I am trying to train the speakernet model. However, I am facing an issue while training with the speaker model.

Error is TypeError: __init__() got an unexpected keyword argument '_metadata'

![Screenshot 2022-11-09 at 14 06 33](https://user-images.githubusercontent.com/85401420/200851335-09f2de40-a262-4c97-912d-9cd51f9f57d7.png)

Could you please help with that.

**Comment by titu1994** [comment]
What Nemo version and pytorch version are you on ? This should be a issue not a discussion really

**Reply by pranayvadd** [reply]
Thanks for reverting.

Currently, I am using Nemo "1.0.0rc1" version and PyTorch "1.7.1" version.

I am not able to find how '_metadata' is passing to '__init__' method.

![image](https://user-images.githubusercontent.com/85401420/200992641-33fb5c11-0862-4b77-bf84-1f9f4faf3644.png)

Could you please help with this? much apperciated !

**Reply by titu1994** [reply]
Too old version. You have to upgrade pytorch Nemo and it's dependencies. Latest version of Nemo is 1.12

**Reply by pranayvadd** [reply]
Thanks for the revert.

I have upgraded to the latest versions of PyTorch and Nemo.

![image](https://user-images.githubusercontent.com/85401420/201096281-8523160d-fdb4-4b35-a897-56aa6be06d9f.png)

Now, I am getting another error "TypeError: __init__() got an unexpected keyword argument 'checkpoint_callback'"

![image](https://user-images.githubusercontent.com/85401420/201099564-270f3668-9575-487a-819a-b87ffd243e60.png)

I checked the "Trainer config", and it is as below:

![Screenshot 2022-11-10 at 12 54 03](https://user-images.githubusercontent.com/85401420/201096906-a71c331b-dfa3-478f-9d84-d0fc5696be3b.png)

But, when I checked in the "https://github.com/Lightning-AI/lightning/blob/master/src/pytorch_lightning/trainer/trainer.py" code.
We don't have any "checkpoint_callback" argument. However, we do have callbacks.

Is there anything I am missing?

Appreciate your responses.

**Reply by titu1994** [reply]
How did you upgrade ? You can do pip install --upgrade nemo_toolkit[all] and it should install correct dependencies

**Reply by pranayvadd** [reply]
Yes... I have installed using nemo_toolkit[all]. Also, I tried with --upgrade as well.

![image](https://user-images.githubusercontent.com/85401420/201340053-0e1a9923-d8d1-45de-af76-82e49a122a69.png)

Still, it is showing the same error.!

**Reply by titu1994** [reply]
Can you print the version of pytorch lightning? If it's greater than 1.7.7, can you explicitly install that ? pip install pytorch_lightning==1.7.7

**Reply by pranayvadd** [reply]
Thanks for the revert.

After installing to 1.7.7 also, it is still showing the same error

![image](https://user-images.githubusercontent.com/85401420/201470815-73ff98a0-30e5-40f8-a4d5-708c1984dd88.png)

Is there anything I am missing

Apperciate your help!

**Reply by titu1994** [reply]
Ok can you print out the full config.trainer and match of to any of the trainer configs inside of the conf dir or the tutorials 

At this point I don't know what's the problemv
